src/SimulationCSharpClient/Client/SimulationClient.cs

```diff
@@ -23,14 +23,16 @@ public partial class SimulationClient
         /// <param name="secret">client secret</param>
         /// <param name="audience">client audience</param>
         /// <param name="httpClient">injected http client</param>
-        public SimulationClient(string baseUrl, string auth0TokenURL, string client, string secret, string audience, HttpClient httpClient = null)
+        /// <param name="maxRetries">http maxRetries.</param>
+        /// <param name="handler">http client handler.</param>
+        public SimulationClient(string baseUrl, string auth0TokenURL, string client, string secret, string audience, HttpClient httpClient = null, int maxRetries = 6, HttpClientHandler handler = null)
```

Comment:
does it need to receive in a HttpClientHandler? would most calls to this client pretty much always use the same/similar handler? i think at a minimum you can check if it is null and provide a default if it is not provided

Reply:
We are just passing this handler to HttpRetryMessageHandler and where it is creating **HttpClientHandler()** object if it is null

`public HttpRetryMessageHandler(int maxRetries = 6, HttpClientHandler handler = null) : base(handler ?? new HttpClientHandler())`